### PR TITLE
Fleetqa/142 cabundle test

### PR DESCRIPTION
**Pull request by mmartin24**
Implements https://app.qase.io/case/FLEET-142

## Done
- Added test to create gitrepo with ca bundle
- Checks successful git repo and related secret creation
- Upon gitrepo deletion, checks ca bundle secret is also deleted.

### CODE REVIEW

File: tests/cypress/e2e/unit_tests/p0_fleet.spec.ts [hunk 1]
```diff
@@ -163,3 +163,53 @@ describe('Test Fleet deployment on PRIVATE repos using KNOWN HOSTS', { tags: '@p
...
+A1UEBhMCR0IxEzARBgNVBAgTClNvbWUtU3RhdGUxFD..AoTC0Jlc3QgQ0Eg
+THRkMTcwNQYDVQQLEy5DbGFzcyAxIFB1YmxpYyBQcm..ENlcnRpZmljYXRp
+b24gQXV0aG9yaXR5MRQwEgYDVQQDEwtCZXN0IENBIE..DAMBgNVHRMEBTAD
+AQH/MA0GCSqGSIb3DQEBBAUAA4IBAQC1uYBcsSncwA..DCsQer772C2ucpX
+xQUE/C0pWWm6gDkwd5D0DSMDJRqV/weoZ4wC6B73f5..bLhGYHaXJeSD6Kr
+XcoOwLdSaGmJYslLKZB3ZIDEp0wYTGhgteb6JFiTtn..sf2xdrYfPCiIB7g
+BMAV7Gzdc4VspS6ljrAhbiiawdBiQlQmsBeFz9JkF4..b3l8BoGN+qMa56Y
+It8una2gY4l2O//on88r5IWJlm1L0oA8e4fR2yrBHX..adsGeFKkyNrwGi/
+7vQMfXdGsRrXNGRGnX+vWDZ3/zWI0joDtCkNnqEpVn..HoX
+-----END CERTIFICATE-----`;
```
Comment (sbulage) [inline tests/cypress/e2e/unit_tests/p0_fleet.spec.ts:198]: It would be nice to have this in `tls.crt` file.

**Review by sbulage — APPROVED** [review]
Small suggestion, rest good to me :+1: 
Thanks :+1:

File: tests/cypress/e2e/unit_tests/p0_fleet.spec.ts (hunk 1 above)
Comment (mmartin24) [reply tests/cypress/e2e/unit_tests/p0_fleet.spec.ts:198]: I guess you mean a `pem` file?
That was my initial idea on the [first](https://github.com/rancher/fleet-e2e/pull/204/commits/e6ca4648443deb407b9d1a77b21c0f50b6289148#diff-9576d8626747e20d3bb4b45d6dfe2773da83d9fec578da7cbc41ea1aff285b0b) commit.
However I was not able to translate it into typing.
I know how to attach it, but not how to read it a and write it.
If you can help out with something that you know it works, I will happily change it.

Comment (sbulage) [reply tests/cypress/e2e/unit_tests/p0_fleet.spec.ts:198]: Okay.

Comment (mmartin24) [reply tests/cypress/e2e/unit_tests/p0_fleet.spec.ts]: Managed to find out how.
Addressed in commit `4ac25415c8e919e01ca5b8e9b7126b23966cf1e1`
Thanks for the extra push here.

**Comment by mmartin24** [comment]
Last CI got some hiccups but everything turned out green after another one: https://github.com/rancher/fleet-e2e/actions/runs/11013553408/job/30582212046

**Pull request merged**